PS-9601 [DOCS] - Release notes for 5.7.44-53 (#459)
percona/psmysql-docs: docs/release-notes/5.7.44-53.md

# Percona Server for MySQL 5.7.44-53 (2025-02-05)

--8<--- "eol.release.md"

## Release Information  

Percona Server for MySQL 5.7.44-53 builds on MySQL 5.7.44 Community Edition,  
adding enterprise-grade features and security enhancements from Percona.  

## Bug fixes

> * [Bug#36234681](https://github.com/mysql/mysql-server/commit/a8c96ce81ab): A `DELETE` operation on a  table with a self-referential foreign key contraint involving a full-text index triggered an assertion. 

Reviewer: ```suggestion
* [Bug#36234681](https://github.com/mysql/mysql-server/commit/a8c96ce81ab): A `DELETE` operation on a  table with a self-referential foreign key constraint involving a full-text index triggered an assertion. 
```

* [Bug#37189985](https://github.com/mysql/mysql-server/commit/0fe52ea5c8f): Using `ALTER TABLE ... ALGORITHM=INPLACE` may result a in server exits and corrupt spatial indexes when used on a table with an auto-increment column.

* [Bug#33691659](https://github.com/mysql/mysql-server/commit/551288baa7a): Performing `ON DELETE CASCADE` on a table with a secondary index on a generated column may cause a server exit in `innobase_get_computed_value`.

* [Bug#33327093](https://github.com/mysql/mysql-server/commit/0ac176453bf): An operation attempted to update a virtual column while building an update node for a child table and caused an assertion failure in
`row_MySQL_pad_col`.

> * [Bug#34574604](https://github.com/mysql/mysql-server/commit/632ca5cfe1b): The virual column information for a row containing an externally stored BLOB was not always logged during an `UPDATE` operation and may cause permanent corruptions.

Reviewer: ```suggestion
* [Bug#34574604](https://github.com/mysql/mysql-server/commit/632ca5cfe1b): The virtual column information for a row containing an externally stored BLOB was not always logged during an `UPDATE` operation and may cause permanent corruptions.
```

* [Bug#36593235](https://github.com/mysql/mysql-server/commit/9a32e4495c5): A subquery containing an aggregate function `WITH ROLLUP` which was part of a row value comparator was not always processed correctly.

* [Bug#37068527](https://github.com/mysql/mysql-server/commit/04b8be52e59): When the server is compiled with XCode 16, a bad comparator caused a server exit during the mysqld initialization. 

* [Bug#37132323](https://github.com/mysql/mysql-server/commit/dd8b0c5c7d6): The server exited because the database cache wasn't flushed when a user was dropped.

* [Bug#37063288](https://github.com/mysql/mysql-server/commit/b76c48b031d): The server exited because AppArmor denied access to the file `/proc/<pid>/task/<thread_id>/mem` which is needed to generate a stack trace.

* [Bug#36615714](https://github.com/mysql/mysql-server/commit/6e08d4427fc): Fixes a buffer overrun in my_print_help.

* [Bug#30875669](https://github.com/mysql/mysql-server/commit/c510fe1470a): The mysql client now allows to send a query containing a number sign or double dash in an optimizer hint comment.

> * [Bug#36879147](https://github.com/mysql/mysql-server/commit/a6b6a6b9861%7Chttps://g): Adds support to 64 bit `FTS_DOC_ID` doc ids for InnoDB FULLTEXT index where earlier, they were imprecisely limited to the maximum unsigned 32-bit value.

Reviewer: ```suggestion
* [Bug#36879147](https://github.com/mysql/mysql-server/commit/a6b6a6b9861): Adds support to 64 bit `FTS_DOC_ID` doc ids for InnoDB FULLTEXT index where earlier, they were imprecisely limited to the maximum unsigned 32-bit value.
```

> Percona Server for MySQL 5.7.44-53 (2025-02-05)

Reviewer 2: ```suggestion
# Percona Server for MySQL 5.7.44-53 (2025-02-05)
```

--8<--- "eol.release.md"

## Release Information  

Percona Server for MySQL 5.7.44-53 builds on MySQL 5.7.44 Community Edition,  
adding enterprise-grade features and security enhancements from Percona.  

## Bug fixes

> * [Bug#36234681](https://github.com/mysql/mysql-server/commit/a8c96ce81ab): InnoDB: The server encounters a failed assertion where the result does not equal
> `FTS_INVALID`. This indicates an unexpected state in the code, requiring
> attention to prevent further errors.

Reviewer: ```suggestion
* [Bug#36234681](https://github.com/mysql/mysql-server/commit/a8c96ce81ab): A DELETE operation on a table with a self referential foreign key involving a full-text index triggers an assertion. 
```

* [Bug#37189985](https://github.com/mysql/mysql-server/commit/0fe52ea5c8f): Using `INPLACE ALTER TABLE` results in server exits and corrupts spatial
indexes. This issue affects data integrity and must be fixed to prevent

> similar failures in future operations.

Reviewer: ```suggestion
* [Bug#37189985](https://github.com/mysql/mysql-server/commit/0fe52ea5c8f): Using ALTER TABLE ... ALGORITHM=INPLACE may result in server exit and corrupt spatial indexes when used on a table with auto increment column.
```

> * [Bug#33691659](https://github.com/mysql/mysql-server/commit/551288baa7a): When performing `ON DELETE CASCADE` with a generated column, the server exits
> in `innobase_get_computed_value`.

Reviewer: ```suggestion
[Bug#33691659](https://github.com/mysql/mysql-server/commit/551288baa7a): Performing `ON DELETE CASCADE` on a table with a secondary index on a generated column may cause a server exit in in `innobase_get_computed_value`.
```

> * [Bug#33327093](https://github.com/mysql/mysql-server/commit/0ac176453bf): A foreign key (FK) constraint causes an assertion failure in
> `row_MySQL_pad_col`, leading to a server exit.

Reviewer: ```suggestion
* [Bug#33327093](https://github.com/mysql/mysql-server/commit/0ac176453bf): An update operation attempted to update a virtual column while building an update node for a child table causes an assertion failure in `row_MySQL_pad_col`
```

> * [Bug#34574604](https://github.com/mysql/mysql-server/commit/632ca5cfe1b): The server experienced an exit due to `ERROR 1712 (HY000): Index PRIMARY was corrupted`.

Reviewer: ```suggestion
* [Bug#34574604](https://github.com/mysql/mysql-server/commit/632ca5cfe1b): Virtual column information for a row containing an externally stored BLOB was not always logged during an UPDATE operation which may cause permanent corruptions.
```

> * [Bug#36593235](https://github.com/mysql/mysql-server/commit/9a32e4495c5): "The MySQL server version 8.3.0 experienced a server exit at `Item_rollup_sum_switcher::current_arg`.

Reviewer 2: ```suggestion
* [Bug#36593235](https://github.com/mysql/mysql-server/commit/9a32e4495c5): The MySQL server version 8.3.0 experienced a server exit at `Item_rollup_sum_switcher::current_arg`.
```

> * [Bug#36593235](https://github.com/mysql/mysql-server/commit/9a32e4495c5): "The MySQL server version 8.3.0 experienced a server exit at `Item_rollup_sum_switcher::current_arg`.

Reviewer: ```suggestion
* [Bug#36593235](https://github.com/mysql/mysql-server/commit/9a32e4495c5): A subquery containing an aggregate function WITH ROLLUP which was part of a row value comparator was not always processed correctly. 
```

> * [Bug#37068527](https://github.com/mysql/mysql-server/commit/04b8be52e59): When they compiled with XCode 16, a bad comparator caused a server exit during the mysqld initialization. 

Reviewer: ```suggestion
* [Bug#37068527](https://github.com/mysql/mysql-server/commit/04b8be52e59): When the server is compiled with XCode 16, a bad comparator caused a server exit during the mysqld initialization. 
```

* [Bug#37132323](https://github.com/mysql/mysql-server/commit/dd8b0c5c7d6): The server exited because the database cache wasn't flushed when a user was dropped.

> * [Bug#37063288](https://github.com/mysql/mysql-server/commit/b76c48b031d): The server exited because AppArmor denied access to the files `/proc/self/task/<tid>/mem`.

Reviewer: ```suggestion
* [Bug#37063288](https://github.com/mysql/mysql-server/commit/b76c48b031d): The server exited because AppArmor denied access to the file `/proc/<pid>/task/<thread_id>/mem` which is needed to generate a stack trace.
```

> * [Bug#36615714](https://github.com/mysql/mysql-server/commit/6e08d4427fc): "The server exited because of a buffer overrun in my_print_help.

Reviewer 2: ```suggestion
* [Bug#36615714](https://github.com/mysql/mysql-server/commit/6e08d4427fc): The server exited because of a buffer overrun in my_print_help.
```

> * [Bug#36615714](https://github.com/mysql/mysql-server/commit/6e08d4427fc): "The server exited because of a buffer overrun in my_print_help.

Reviewer: ```suggestion
* [Bug#36615714](https://github.com/mysql/mysql-server/commit/6e08d4427fc): Fixes a buffer overrun in my_print_help.
```

> * [Bug#30875669](https://github.com/mysql/mysql-server/commit/c510fe1470a): The server exited because the core client couldn't send a query containing a number sign or double dash in a hint comment.

Reviewer: ```suggestion
* [Bug#30875669](https://github.com/mysql/mysql-server/commit/c510fe1470a): The mysql client now allows to send a query containing a number sign or double dash in an optimizer hint comment.
```

> * [Bug#36879147](https://github.com/mysql/mysql-server/commit/a6b6a6b9861%7Chttps://g): "The server exited because the InnoDB FULLTEXT index limited `FTS_DOC_ID` to the maximum unsigned 32-bit value.

Reviewer 2: ```suggestion
* [Bug#36879147](https://github.com/mysql/mysql-server/commit/a6b6a6b9861%7Chttps://g): The server exited because the InnoDB FULLTEXT index limited `FTS_DOC_ID` to the maximum unsigned 32-bit value.
```

> * [Bug#36879147](https://github.com/mysql/mysql-server/commit/a6b6a6b9861%7Chttps://g): "The server exited because the InnoDB FULLTEXT index limited `FTS_DOC_ID` to the maximum unsigned 32-bit value.

Reviewer: ```suggestion
* [Bug#36879147](https://github.com/mysql/mysql-server/commit/a6b6a6b9861): Adds support to 64 bit `FTS_DOC_ID` doc ids for InnoDB FULLTEXT index where earlier, they were imprecisely limited to the maximum unsigned 32-bit value.
```
